### DISCUSSION TITLE
setup auto scan folder and ocr new pdf

### DISCUSSION THREAD

**Discussion by BeckeBauer** [Stirling-Tools/Stirling-PDF#989] [Q&A] [answered]
First of all, great app 👍
I just noticed the option `Local Directory Scanning and File Processing` and tried to set it up. Since I am running the app via docker, I mounted my local scan folder under `./pipeline/watchedFolders`. 

However, I am lost when it comes to creating a JSON configuration file. I used the following configuration which I was able to extract from the pipeline feature:
```
{
  "name": "ocr",
  "pipeline": [
    {
      "operation": "/api/v1/misc/ocr-pdf",
      "parameters": {
        "fileInput": "automated"
      }
    }
  ],
  "_examples": {
    "outputDir": "{outputFolder}/{folderName}",
    "outputFileName": "{filename}-{pipelineName}-{date}-{time}"
  },
  "outputDir": "{outputFolder}",
  "outputFileName": "{filename}"
}
```
Unfortunately, clicking on the wheel with yellow background to configure the pipeline settings does not do anything in my case, so I have no idea how the parameter (in particular changing ocr language) could look like.
 
Also, how can I define that the "ocr-ed" files are saved in the same directory they are in?

Is there anything that needs to be added to the configuration?

Any sample JSON configuration file or help is appreciated

**Comment by Frooodle** [comment]
seems to be a bug in latest release, Will try patch soon!

**Comment by BeckeBauer** [comment]
Okay, great but I am kind of confused. Do you refer to the issue that the settings wheel in the pipeline feature  is not working? 

If so, I was wondering if you could help me with the creation of the JSON config file (configuration of the output folder)?

**Reply by Frooodle** [reply]
In latest update i will release soon it should create a better config file for you which will work

**Reply by Frooodle** [reply]
please update and try recreate the config!

**Reply by BeckeBauer** [reply]
After updating, JSON config files can be created with the pipeline feature 👍. For ocr, I got the following config:
```
{
  "name": "ocr",
  "pipeline": [
    {
      "operation": "/api/v1/misc/ocr-pdf",
      "parameters": {
        "languages": "deu, eng",
        "sidecar": false,
        "deskew": false,
        "clean": false,
        "cleanFinal": false,
        "ocrType": "skip-text",
        "ocrRenderType": "hocr",
        "removeImagesAfter": false,
        "fileInput": "automated"
      }
    }
  ],
  "_examples": {
    "outputDir": "{outputFolder}/{folderName}",
    "outputFileName": "{filename}-{pipelineName}-{date}-{time}"
  },
  "outputDir": "{outputFolder}",
  "outputFileName": "{filename}"
}
```

The configuration works when I use it to ocr pdf-files in the browser. So, I saved the ocr.json in the folder `./pipeline/watchedFolders/Scans` and put some pdf-files into the folder but nothing happens. 

Is there anything I need to do to activate the feature `Local Directory Scanning and File Processing`?

**Reply by Frooodle** [reply]
Weird that should work
Let me try reproduce for you

**Reply by Frooodle** [reply, marked as answer]
One more fix coming, will be released in about 10 minutes, 0.22.6

**Reply by Frooodle** [reply]
If you update it should start automatically working now :)

**Reply by BeckeBauer** [reply]
Now it works, thank you 👌

**Reply by BeckeBauer** [reply]
If someone is interested: I created two subfolders under my folder `Scan ` (`scan_in` and `scan_ocr`) and mounted both folders via docker
```
- /path to/Scans/scans_in:/pipeline/watchedFolders/scans_in 
- /path to/Scans/scans_ocr:/pipeline/finishedFolders/scans_ocr
```
If you now place your pdfs into the scan_in folder and wait a bit, you can find them in scans_ocr as searchable pdf after some time
